pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -98,13 +102,31 @@ func getKubeClient(timeout time.Duration) (kubernetes.Interface, error) {
 	return kubernetes.NewForConfig(clientConfig)
 }
 
+func getInformerFactory() (informers.SharedInformerFactory, error) {
+	timeoutSeconds := time.Duration(config.Datadog.GetInt64("kubernetes_informers_restclient_timeout"))
+	resyncPeriodSeconds := time.Duration(config.Datadog.GetInt64("kubernetes_informers_resync_period"))
+	client, err := getKubeClient(timeoutSeconds * time.Second)
+	if err != nil {
+		log.Infof("Could not get apiserver client: %v", err)
+		return nil, err
+	}
+	return informers.NewSharedInformerFactory(client, resyncPeriodSeconds*time.Second), nil
+}
+
 func (c *APIClient) connect() error {
 	var err error
 	c.Cl, err = getKubeClient(time.Duration(c.timeoutSeconds) * time.Second)
 	if err != nil {
 		log.Infof("Could not get apiserver client: %v", err)
 		return err
 	}
+
+	// informer factory uses its own clientset with a larger timeout
+	c.InformerFactory, err = getInformerFactory()
+	if err != nil {
+		return err
+	}
```

Comment:
All node agents would create the informer factory but only the DCA actually calls `InformerFactory.Start()`.

It might make more sense to stop trying to share the `APIClient` type between both systems.